features/dd-sdk-android-logs/src/test/kotlin/com/datadog/android/log/internal/domain/DatadogLogGeneratorTest.kt

```diff
@@ -377,6 +377,103 @@ internal class DatadogLogGeneratorTest {
         assertThat(log.additionalProperties).doesNotContainKey(LogAttributes.SOURCE_TYPE)
     }
 
+    @Test
+    fun `M not add fingerprint W creating the Log { fingerprint attribute not set }`() {
+        // WHEN
+        val log = testedLogGenerator.generateLog(
+            fakeLevel,
+            fakeLogMessage,
+            fakeThrowable,
+            fakeAttributes,
+            fakeTags,
+            fakeTimestamp,
+            fakeThreadName,
+            fakeDatadogContext,
+            attachNetworkInfo = true,
+            fakeLoggerName
+        )
+
+        // THEN
+        assertThat(log).hasError(
+            LogEvent.Error(
+                kind = fakeThrowable.javaClass.canonicalName,
+                stack = fakeThrowable.stackTraceToString(),
+                message = fakeThrowable.message,
+                sourceType = null,
+                fingerprint = null,
+                threads = null
+            )
+        )
+    }
+
+    @Test
+    fun `M add fingerprint W creating the Log { fingerprint attribute set }`() {
+        // WHEN
+        val modifiedAttributes = fakeAttributes.toMutableMap().apply {
+            put(LogAttributes.ERROR_FINGERPRINT, "fake_fingerprint")
+        }
```

Comment:
minor: it is better to use a random string. same for the test below.

```suggestion
        // WHEN
        val fakeFingerprint = forge.aString()
        val modifiedAttributes = fakeAttributes.toMutableMap().apply {
            put(LogAttributes.ERROR_FINGERPRINT, fakeFingerprint)
        }
```